rosdoc2/verbs/build/builders/sphinx_builder.py

```diff
@@ -78,23 +78,22 @@ def ensure_global(name, default):
     from exhale import utils
     exhale_args.update({{
         # These arguments are required.
-        "containmentFolder": "{user_sourcedir}/api",
-        "rootFileName": "library_root.rst",
-        "rootFileTitle": "{package_name} API",
+        "containmentFolder": "{user_sourcedir}/generated",
+        "rootFileName": "index.rst",
         "doxygenStripFromPath": "..",
         # Suggested optional arguments.
         "createTreeView": True,
+        "fullToctreeMaxDepth": 1,
+        "unabridgedOrphanKinds": [],
+        "fullApiSubSectionTitle": "Reference",
         # TIP: if using the sphinx-bootstrap-theme, you need
         # "treeViewIsBootstrap": True,
         "exhaleExecutesDoxygen": False,
         # Maps markdown files to the "md" lexer, and not the "markdown" lexer
         # Pygments registers "md" as a valid markdown lexer, and not "markdown"
         "lexerMapping": {{r".*\.(md|markdown)$": "md",}},
-        # This mapping will work when `exhale` supports `:doxygenpage:` directives
-        # Check https://github.com/svenevs/exhale/issues/111
-        # TODO(aprotyas): Uncomment the mapping below once the above issue is resolved.
-        # "customSpecificationsMapping": utils.makeCustomSpecificationsMapping(
-        #     lambda kind: [":project:", ":path:", ":content-only:"] if kind == "page" else []),
+        "customSpecificationsMapping": utils.makeCustomSpecificationsMapping(
```

Comment:
Did we ever resolve @aprotyas issue with this?  Quoting:

```
IIRC, this also changes the behavior when there is a different (not :doxygenpage:) directive. As in, other directives such as :doxygenfunction: wouldn't render correctly with this mapping, and I think you have to provide mappings for each such directive. I will confirm whether that's true in a future comment.
```

Reply:
It's funny, `exhale` documentation is misleading here. If `utils.makeCustomSpecificationsMapping` yields an empty list for the specs of given directive, it'll use Breathe defaults, which are not the same as Exhale defaults (see https://github.com/svenevs/exhale/blob/bbb5d71f126d1281a8c4f1729c218c236f44457b/exhale/utils.py#L185). I can change this to match Exhale defaults though.

Reply:
It's not clear to me whether we should be going for the Breathe or Exhale defaults in that case.  @hidmic @aprotyas I would take whatever your recommendation here is.

Reply:
The Exhale defaults are not bad, but those defaults are only for the `class` and `struct` kinds.
It would be nice to pay the one-time cost of a `rosdoc2_specsMapping` dictionary that lets us control the directive of all kinds that are of interest. It also makes for more extensibility:

```py
rosdoc2_specsMapping = {                                                                               
    'page': [':content-only:'],                  
    **dict.fromkeys(['class', 'struct'], [':members:', ':protected-members:', ':undoc-members:']),  
    # other kind-specs mapping                                                                         
}       

"customSpecificationsMapping": utils.makeCustomSpecificationsMapping(
    lambda kind: rosdoc2_specsMapping.get(kind, [])),
```

Reply:
That looks reasonable to me. See befbade175adfc6369f02be17718d570203f9499.